Modules/Core/Common/test/itkSpatialOrientationTest.cxx

```diff
@@ -0,0 +1,114 @@
+/*=========================================================================
+ *
+ *  Copyright NumFOCUS
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0.txt
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *=========================================================================*/
+
+#include "itkSpatialOrientation.h"
+#include <iostream>
+#include <set>
+
+
+int
+itkSpatialOrientationTest(int, char *[])
+{
+
+  // Test streaming enumeration for SpatialOrientationEnums::CoordinateTerms elements
+  const std::set<itk::SpatialOrientationEnums::CoordinateTerms> allCoordinateTerms{
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_UNKNOWN,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Right,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Left,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Posterior,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Anterior,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Inferior,
+    itk::SpatialOrientationEnums::CoordinateTerms::ITK_COORDINATE_Superior
+  };
+  for (const auto & ee : allCoordinateTerms)
+  {
+    std::cout << "STREAMED ENUM VALUE SpatialOrientationEnums::CoordinateTerms: " << ee << std::endl;
```

Comment:
Is it really helpful to print all these enums to standard output? Or could they also be inserted to a string-stream, instead? For example as follows:

    std::ostringstream stringStream;
    stringStream << ee;
    ASSERT( ! stringStream.str().empty() );

Doing so might reduce the "noise" in the test output.

Reply:
@N-Dekker
- The serialization test not being quantitative has been mentioned before (see the `Additional Information` in #2925). 
- As said, any change to these affects any other strongly typed enums in the code base, and there are a few dozens. The above issue can host any discussion and proposals to improve these. Changes should be done for all strongly typed enum serialization tests at once. So it is beyond the scope of this PR.